src/js/components/TaskListItemComponent.jsx

```diff
@@ -60,6 +18,110 @@ var TaskListItemComponent = React.createClass({
     taskHealthMessage: React.PropTypes.string
   },
 
+  getHostAndPorts: function () {
+    var task = this.props.task;
+    var ports = task.ports;
+
+    if (ports == null || ports.length === 0 ) {
+      return (<span className="text-muted">{task.host}</span>);
+    }
+
+    if (ports != null && ports.length === 1) {
+      return (
+        <a className="text-muted" href={`//${task.host}:${ports[0]}`}>
+          {`${task.host}:${ports[0]}`}
+        </a>
+      );
+    }
+
+    if (ports != null && ports.length > 1) {
+      let portNodes = ports.map(function (port, index) {
+        return (
+          <span key={port}>
+            <a className="text-muted" href={`//${task.host}:${port}`}>
+              {port}
+            </a>
+            {index < ports.length - 1 ? ", " : ""}
+          </span>
+        );
+      });
+
+      return (
+        <span className="text-muted">
+          {task.host}:[{portNodes}]
+        </span>
+      );
+    }
+  },
+
+  getEndpoints: function () {
+    var task = this.props.task;
+
+    if (task.ipAddresses == null) {
+      return this.getHostAndPorts();
+    }
+
+    return this.getServiceDiscoveryEndpoints();
+  },
+
+  getServiceDiscoveryEndpoints: function () {
+    var props = this.props;
+    var task = props.task;
+    var app = AppsStore.getCurrentApp(props.appId);
+
+    if (app.ipAddress != null &&
+        app.ipAddress.discovery != null &&
+        app.ipAddress.discovery.ports != null &&
+        task.ipAddresses != null &&
+        task.ipAddresses.length > 0) {
+
+      let serviceDiscoveryPorts = app.ipAddress.discovery.ports;
+
+      let endpoints = task.ipAddresses.map((address, i) => {
+        let ipAddress = address.ipAddress;
+        let trailingComma = i < task.ipAddresses.length - 1;
+
+        if (serviceDiscoveryPorts.length === 1) {
+          let port = serviceDiscoveryPorts[0].number;
+          return (
+            <span className="text-muted" key={`${ipAddress}:${port}`}>
+              <a className="text-muted"
+                  href={`//${ipAddress}:${port}`}>
+                {`${ipAddress}:${port}`}
+              </a>
+              {trailingComma ? ", " : ""}
```

Comment:
Let's write a helper for this, to get rid of the repeatedly used ```span```-wraper and tenerary operator .

```javascript
  function joinNodes(nodes, seperator) {
    var lastIndex = nodes.length - 1;
    return nodes.map((node, i) => {
      if (lastIndex === i) {
        seperator = null;
      }
      return <span className="text-muted">{node}{seperator}</span>;
    });
  }
```
then you can get rid of this tralingComma-lines completely.